src/CMakeLists.txt

```diff
@@ -871,16 +871,14 @@ add_subdirectory(compressor)
 
 add_subdirectory(tools)
 
-# dmClock
-
-add_subdirectory(dmclock) # after gmock
-add_dependencies(tests dmclock-tests dmclock-data-struct-tests)
+# dmClock (after gmock)
 
+add_subdirectory(dmclock/src)
 if(WITH_TESTS)
-  install(PROGRAMS
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/dmclock-tests
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/dmclock-data-struct-tests
-    DESTINATION bin)
+  # note: add_test is not being called, so dmclock tests aren't part
+  # of ceph tests
+  add_subdirectory(dmclock/test)
+  add_subdirectory(dmclock/support/test)
 endif(WITH_TESTS)
```

Comment:
The purpose of `WITH_TESTS` is not to conditionalize the cmake tests - it's conditionalizing the build of binaries packaged in the `ceph-test` RPM/deb (i.e. the binaries you are dropping in this PR). 

(See https://github.com/ceph/ceph/blob/master/ceph.spec.in#L22 and https://github.com/ceph/ceph/blob/master/ceph.spec.in#L24 and check out how these bconds are used.)